psl/psl-core/src/configuration/configuration_struct.rs

```diff
@@ -161,4 +191,22 @@ impl Configuration {
 
         Ok(())
     }
+
+    pub fn generators_with_files(&self) -> impl Iterator<Item = parser_database::InFile<&Generator>> {
+        self.generators
+            .iter()
+            .zip(&self.generators_files)
+            .map(|(gen, file_id)| (*file_id, gen))
+    }
+
+    pub fn datasources_with_files(&self) -> impl Iterator<Item = parser_database::InFile<&Datasource>> {
```

Comment:
Doesn't `Datasource` already have  a link to a file it defined in? `span().file_id` should be it